content/PreparedStatement/PreparedStatement.q.md

```diff
@@ -0,0 +1,1677 @@
+
+- It creates the multidimensional array named `$data` with two keys `student1` and `student2`.
+
+- It accesses the elements of `$data` using the `array()` construct and the `[]` short array syntax.
+
+
+/// type=SS, answer=[3]
+
+In `student.php` what does `prepare()` method do on line 10?
+
+- It returns an array.
+
+- It executes the SQL statement.
+
+- It prepares the SQL statement for execution.
+
+- It contains the result set of the remaining rows.
+
+- It substitutes the input values of the SQL statement.
+
+
+/// type=SS, answer=[3]
+
+In `student.php`, what is the return value of `$conn->prepare("INSERT INTO students (first_name, last_name, birth_date, gender) VALUES (?,?,?,?)")` on line 10?
+
+- The SQL statement.
+
+- The DSN parameters.
+
+- The `PDOStatment` object.
+
+- The values of the varibale array `$data`.
+
+- The elements of the variable array `$data`.
+
+
+/// type=SS, answer=[3]
+
+In `student.php`, what does `$pstmt` do on line 10?
+
+- It stores the SQL statement.
+
+- It executes the SQL statement.
+
+- It holds the `PDOStatement` object.
+
+- It inserts values into the `students` table.
+
+- It stores the values of the `students` table.
+
+
+/// type=SS, answer=[3]
+
+In `student.php`, which of the following represents the positional placeholder as a parameter on line 10?
+
+- `students`
+
+- `first_name`
+
+- `(?,?,?,?)`
+
+- `INSERT INTO`
+
+- `(first_name, last_name, birth_date, gender)`
+
+
+/// type=SS, answer=[5]
+
+In `student.php`, what does the positional placeholder do `VALUES (?,?,?,?)` on line 10?
+
+- It removes question mark values in the SQL statement.
+
+- It inserts question marks as values to the `students` table.
+
+- It sets random values of question marks to the SQL statement.
+
+- It replaces the values in the SQL statement with question marks to have random values.
+
+- It substitutes the values in the SQL statement with question marks to prevent SQL injection attacks.
+
+
+/// type=MS, answer=[2,3,4,5]
+
+In `student.php`, which statements correctly describe `$pstmt = $conn->prepare("INSERT INTO students (first_name, last_name, birth_date, gender) VALUES (?,?,?,?)");` on line 10?
+
+- It executes the SQL statement.
+
+- It prepares the SQL statement for execution.
+
+- It returns the result set as a `PDOStatement` object.
+
+- It calls the `prepare()` method of the `$conn` object.
+
+- It assigns the `PDOStatement` object to the variable `$pstmt`.
+
+
+/// type=MS, answer=[3,5]
+
+Which statements correctly describe the `foreach` construct on line 11 of `student.php`?
+
+- It displays all elements in `$row`.
+
+- It accesses through each element in `$data`.
+
+- It iterates through each element in the array.
+
+- It creates and initializes the array variable `$data`.
+
+- It assigns the value of the current element on each iteration to `$row`.
+
+
+/// type=SS, answer=[2]
+
+In `student.php`, what does `execute()` method do on line 12?
```

Comment:
Basic grammar error here...

Missing article `the` before `execute()`

`execute()` method ---> __the__ `execute()` method